langchain/src/document_loaders/web/notionapi.ts

```diff
@@ -11,44 +15,118 @@ import type {
   MdBlock,
 } from "notion-to-md/build/types";
 
-import { BaseDocumentLoader } from "../base.js";
+import Bottleneck from "bottleneck";
```

Comment:
We already have a concurrency library:

https://www.npmjs.com/package/p-queue

Would prefer to use that if possible instead of adding a new peer dep?

Reply:
I will look into it, the preference is always to reduce dependencies. 👍

Reply:
You can use it with this util class we have actually, which might be even easier:

https://github.com/hwchase17/langchainjs/blob/main/langchain/src/util/async_caller.ts

Reply:
I have replaced bottleneck with the AsyncCaller utility and it appears to be working without issue.